### ISSUE TITLE
Cannot stub where protocol function has multiple arities

### ISSUE THREAD

**Issue by joelittlejohn** [bguthrie/shrubbery#9]
This kind of thing fails:

```clj
(defprotocol AProtocol
  (qux [this] [this that]))

...

  (testing "with multiple arities"
    (let [subject (stub AProtocol {:qux 1})]
      (is (= 1 (qux subject)))
      (is (= 1 (qux subject :that)))))
```

because stub implementations are only added for the first arglist in the signature.

**Comment by bguthrie** [comment]
Yup, that's a problem. I didn't support this initially because I wasn't sure if anyone would ever need it. If you've got a use case, I'm happy to fix.

**Comment by joelittlejohn** [comment]
I had to bail out of a test I was trying to write recently due to this so I'd love to see it added.

Another thing I've been thinking about recently, which is somewhat related (although this may go too far in changing the spirit of shrubbery) is whether we could change the signature of `stub` entirely to allow arg values to be given when setting up a stubbed function. When the arg values match, the return value would be returned.

Obviously it's possible to use `received?` afterwards to match/very args given, but it would be more useful to supply these when subbing IMO.

How do you feel about that?

Happy to have a go at implementing this but I'd welcome your thoughts on what alternative signature for `stub` would be preferred.

**Comment by bguthrie** [comment]
This issue has been fixed by 1a65b4f2bf72f7dc6fb9dc5f9f7f31b0ae5fe2d7.

As for the `stub` question, I'm open to taking a look. Can you open a new issue, ideally one that includes syntax that you're interested in, and/or an example of usage? We can discuss it there.

**Comment by joelittlejohn** [comment]
@bguthrie oh great, thanks!
